classes/note/NoteDAO.inc.php

```diff
@@ -81,45 +86,55 @@ public function getByUserId($userId, $rangeInfo = null)
      * @param int $orderBy Optional sorting field constant: self::NOTE_ORDER_...
      * @param int $sortDirection Optional sorting order constant: SORT_DIRECTION_...
      *
-     * @return object DAOResultFactory containing matching Note objects
+     * @return LazyCollection<Note>
      */
-    public function getByAssoc($assocType, $assocId, $userId = null, $orderBy = self::NOTE_ORDER_DATE_CREATED, $sortDirection = self::SORT_DIRECTION_DESC, $isAdmin = false)
+    public function getByAssoc(int $assocType, int $assocId, ?int $userId = null, bool $isAdmin = false, 
+        bool $orderData = true, int $orderBy = self::NOTE_ORDER_DATE_CREATED, int $sortDirection = self::SORT_DIRECTION_DESC) : LazyCollection
     {
-        $params = [(int) $assocId, (int) $assocType];
+        $query = DB::table('notes')
+            ->where('assoc_id', '=', $assocId)
+            ->where('assoc_type', '=', $assocType);
+
         if ($userId) {
-            $params[] = (int) $userId;
+            $query->where('user_id', '=', $userId);
         }
 
-        // Sanitize sort ordering
-        switch ($orderBy) {
-            case self::NOTE_ORDER_ID:
-                $orderSanitized = 'note_id';
-                break;
-            case self::NOTE_ORDER_DATE_CREATED:
-            default:
-                $orderSanitized = 'date_created';
+        if (!$isAdmin) {
+            $query->where(function ($query) {
+                $query->whereNotNull('title')
+                    ->orWhereNotNull('contents');
+            });
         }
-        switch ($sortDirection) {
-            case self::SORT_DIRECTION_ASC:
-                $directionSanitized = 'ASC';
-                break;
-            case self::SORT_DIRECTION_DESC:
-            default:
-                $directionSanitized = 'DESC';
+
+        if ($orderData) {
```

Comment:
There's no harm in having a default sort order, so I don't think it's necessary to add an `$orderData` parameter. If it was necessary, it might be better implemented with a new `NOTE_ORDER_NONE` constant. Just to help keep the parameter lists short for these functions! Overall I'd prefer just to keep the old `NOTE_ORDER_DATE_CREATED` default.

Reply:
Removed the switch from the `orderData` parameter from the function and code. Changed `getByAssoc` reference from codebase